pygmt/src/solar.py

```diff
@@ -0,0 +1,99 @@
+"""
+solar - Plot day-night terminators.
+"""
+
+import datetime
+
+import pandas as pd
+from pygmt.clib import Session
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import build_arg_string, fmt_docstring, kwargs_to_strings, use_alias
+
+
+@fmt_docstring
+@use_alias(
+    B="frame",
+    G="fill",
+    J="projection",
+    R="region",
+    U="timestamp",
+    V="verbose",
+    W="pen",
+    X="xshift",
+    Y="yshift",
+    c="panel",
+    p="perspective",
+    t="transparency",
+)
+@kwargs_to_strings(R="sequence", c="sequence_comma", p="sequence")
+def solar(self, terminator="d", terminator_datetime=None, **kwargs):
+    r"""
+    Plot day-light terminators.
+
+    This function plots the day-night terminator. Additionally, it can
+    plot the terminators for civil twilight, nautical twilight, and
+    astronomical twilight.
+
+    Full parameter list at :gmt-docs:`solar.html`
+
+    {aliases}
+
+    Parameters
+    ----------
+    terminator : str
+        Set the type of terminator displayed. Valid arguments are
+        **day_night**, **civil**, **nautical**, and **astronomical**, which
+        can be set with either the full name or the first letter of the name.
+        [Default is **day_night**]
+    terminator_datetime : str or datetime object
+        Set the UTC date and time of the displayed terminator. It can be
+        passed as a string or a datetime object.
+        [Default is the current UTC date and time]
+    {R}
+    {J}
+    {B}
+    fill : str
+        Color or pattern for filling of terminators.
+    pen : str
+        Set pen attributes for lines. The default pen
+        is ``default,black,solid``.
+    {U}
+    {V}
+    {XY}
+    {c}
+    {p}
+    {t}
+    """
+    # TODO: Incorporate "-T+zTZ"
```

Comment:
This is just a note of why we don't support timezone yet (https://github.com/GenericMappingTools/gmt/issues/4695).

Reply:
Unfortunately, pylint doesn't like TODO comments: `pygmt/src/solar.py:67:2: W0511: TODO: Incorporate "-T+zTZ" (fixme)`.

Reply:
OK. Then we can just remove the `TODO` and open a feature request after merging this PR, so that we won't forget it.

Reply:
resolved in d3baca7